holopy/scattering/tests/test_scatteringtheory.py

```diff
@@ -0,0 +1,335 @@
+import unittest
+
+import numpy as np
+import xarray as xr
+from numpy.testing import assert_allclose, assert_equal
+from nose.plugins.attrib import attr
+
+from holopy.core import detector_grid, detector_points
+from holopy.core.metadata import update_metadata, flat
+from holopy.scattering.theory.scatteringtheory import ScatteringTheory
+from holopy.scattering.theory import Mie
+from holopy.scattering.scatterer import Sphere, Spheres, Ellipsoid
+from holopy.scattering.errors import TheoryNotCompatibleError
+from holopy.scattering.calculations import prep_schema
+from holopy.scattering.tests.common import xschema as XSCHEMA
+
+
+SPHERE = Sphere(n=1.5, r=1.0, center=(0, 0, 2))
+SPHERES = Spheres([
+    Sphere(n=1.5, r=1.0, center=(-1, -1, 2)),
+    Sphere(n=1.5, r=1.0, center=(+1, +1, 2)),
+    ])
+ELLIPSOID = Ellipsoid(n=1.5, r=(0.5, 0.5, 1.2), center=(0, 0, 2))
+TOLS = {'atol': 1e-14, 'rtol': 1e-14}
+MEDTOLS = {'atol': 1e-7, 'rtol': 1e-7}
+
+
+SCAT_SCHEMA = prep_schema(
+    detector_grid(shape=(5, 5), spacing=.1),
+    medium_index=1.33, illum_wavelen=0.66, illum_polarization=False)
+
+
+class MockTheory(ScatteringTheory):
```

Comment:
Hmmm.. the point of the MockTheory is just for unit tests, so I don't want to add it to the the scatteringtheory file. My logic behind making the ```_can_handle``` only a ```Sphere``` was that it lets me test that it can't handle some things and fails gracefully, and it lets me test some of the logic for ```Scatterers``` objects.  I'll clarify these with some comments in the docstring of ```MockTheory```.

Also, right now ```ScatteringTheory``` should be the abstract base class object that all scattering theories are derived from, so I don't want to change it to something else. Although right now the it's not enforced, and ```ScatteringTheory``` does not fail gracefully when its ```__init__``` is not overridden. I'll TDD that as well.

Reply:
First one is fixed with dbbbe81

Reply:
But I don't think the second is easily fixable. I would like to change the ```__init__``` to raise a ```NotImplementedError```, as the repr of ```ScatteringTheory``` crashes when the inherited class doesn't have its own ```__init__```. However, many of the subclasses call ```ScatteringTheory.__init__``` through a ```super()``` call, so simply raising a not implemented error breaks all these. 

So I think it's best to just leave it as is for now, unless you disagree.

Reply:
Let's leave it and keep it in mind for the next overhaul.